openedx/core/djangoapps/oauth_dispatch/management/commands/edx_clear_expired_tokens.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import unicode_literals
+
+import logging
+from datetime import timedelta
+
+from django.core.exceptions import ImproperlyConfigured
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from django.utils import timezone
+
+from oauth2_provider.models import AccessToken, Grant, RefreshToken
+from oauth2_provider.settings import oauth2_settings
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    help = "Clear expired access tokens and refresh tokens for DOT"
+
+    def add_arguments(self, parser):
+        parser.add_argument('--batch_size',
+                            action='store',
+                            dest='batch_size',
+                            type=int,
+                            default=1000,
+                            help='Maximum number of database rows to delete per query. '
+                                 'This helps avoid locking the database when deleting large amounts of data.')
+
+    def clear_table_data(self, query_set, batch_size, model):
+        while query_set.exists():
+            qs = query_set[:batch_size]
+            query_set = query_set[batch_size:] if query_set[batch_size:] else query_set
+            history_batch = qs.values_list('id', flat=True)
+            with transaction.atomic():
+                model.objects.filter(pk__in=list(history_batch)).delete()
+
+    def handle(self, *args, **options):
+        batch_size = options['batch_size']
+        now = timezone.now()
+        refresh_expire_at = None
+
+        refresh_token_expire_seconds = oauth2_settings.REFRESH_TOKEN_EXPIRE_SECONDS
+
+        if refresh_token_expire_seconds:
+            if not isinstance(refresh_token_expire_seconds, timedelta):
+                try:
+                    refresh_token_expire_seconds = timedelta(seconds=refresh_token_expire_seconds)
+                except TypeError:
+                    e = "REFRESH_TOKEN_EXPIRE_SECONDS must be either a timedelta or seconds"
+                    raise ImproperlyConfigured(e)
+            refresh_expire_at = now - refresh_token_expire_seconds
+
+        if refresh_expire_at:
```

Comment:
Why are there `if` statements in this management command? Make sure the appropriate settings are set, or fallback to a sane default. This branching is confusing.